### PR TITLE
Fix drag'n drop type checking for NodePaths

### LINKED ISSUE #62691: Drag and drop for NodePaths isn't "type safe"
### Godot version

v4.0.alpha.calinou.e16242b73

### System information

Arch Linux, Vulkan, Intel(R) HD Graphics 520 (SKL GT2)

### Issue description

When drag and dropping nodes from the scene tree to exposed NodePaths (as implemented in #55761), nodes that aren't of the desired type can still be dropped onto the exposed property.

https://user-images.githubusercontent.com/30960698/177061798-d7826840-f637-41d0-a431-5b40d685e387.mp4

Also goes hand in hand with #62483.

### Steps to reproduce

1. Export a NodePath using `@export_node_path` and specify a node type
2. Drag and drop a node of a different type on it

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Nolkaloid**
Fixes #62691

### CODE REVIEW

**Comment by KoBeWi** [comment]
This solution doesn't allow using sub-classes, which is a valid use-case and should be supported.

**Comment by Nolkaloid** [comment]
Updated! It should work for sub-classes now :)

File: editor/editor_properties.cpp [hunk 1]
```diff
@@ -3175,7 +3175,18 @@ bool EditorPropertyNodePath::is_drop_valid(const Dictionary &p_drag_data) const
 		return false;
 	}
 	Array nodes = p_drag_data["nodes"];
-	return nodes.size() == 1;
+	if (nodes.size() != 1) {
+		return false;
+	}
+
+	Node *dropped_node = get_tree()->get_edited_scene_root()->get_node(nodes[0]);
```
Comment (KoBeWi) [inline editor/editor_properties.cpp:3182]: You could add `ERR_FAIL_NULL` here in case the path is invalid for whatever reason.

Comment (Nolkaloid) [reply editor/editor_properties.cpp:3182]: Done! Thanks for the guidance, I'm new to this :)

**Review by KoBeWi — APPROVED** [review]

File: editor/editor_properties.cpp [hunk 2]
```diff
@@ -3175,7 +3175,21 @@ bool EditorPropertyNodePath::is_drop_valid(const Dictionary &p_drag_data) const
...
+	ERR_FAIL_NULL_V_MSG(dropped_node, false,
+			vformat("%s isn't a valid NodePath", nodes[0]));
```
[the remark this answers is not in the document]
Comment (Nolkaloid) [reply editor/editor_properties.cpp:3184]: So I shall replace it with `ERR_FAIL_NULL_V` ?

[the remark this answers is not in the document]
Comment (KoBeWi) [reply editor/editor_properties.cpp:3184]: Yes.

[the remark this answers is not in the document]
Comment (Nolkaloid) [reply editor/editor_properties.cpp]: Done!

**Comment by akien-mga** [comment]
Thanks!

**Comment by KoBeWi** [comment]
This actually breaks the following code:
```GDScript
@export var n: NodePath
```
No node can be dropped to a generic NodePath variable.

**Pull request merged**